server/proxy/incoming.go

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2020-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package proxy
+
+import (
+	"net/http"
+
+	pluginapi "github.com/mattermost/mattermost-plugin-api"
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost-plugin-apps/apps"
+	"github.com/mattermost/mattermost-plugin-apps/server/config"
+	"github.com/mattermost/mattermost-plugin-apps/server/mmclient"
+	"github.com/mattermost/mattermost-plugin-apps/utils"
+	"github.com/mattermost/mattermost-plugin-apps/utils/httputils"
+)
+
+type Incoming struct {
```

Comment:
Out of scope of this PR: I've being thinking about a attaching a logger to this to get rid of the all the log annotation that we do down the stack.

Also, maybe a `context.Context` would be something that could also fit in here (I know, it's against the docs)

Reply:
1/5 `Config` can (should) be a stack, will allow for cascading the loggers, and other stuff.

Reply:
Resolving since out of scope